### PR TITLE
Move architecture handling into its own package

### PR DESCRIPTION
**Pull request by mvo5**
The architecture handling used to be partly in the helpers and
partly in the snappy package. By moving it into a common package
subsequent branches (feature/snapfs-mount2) will become simpler.

Its important that tools like ubuntu-device-flash/uflash can pretend
to be a different architecture when talking to the store and installing
snaps into the newly created image (that might not be equivalent
to the host architecture).

### CODE REVIEW

File: arch/arch.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+func goToUbuntuArchitecture() string {
+	switch goarch {
+	case "386":
+		return "i386"
+	case "arm":
+		return "armhf"
+	default:
+		return goarch
```
Comment (zyga) [inline arch/arch.go]: I'd panic() in this case

**Comment by zyga** [comment]
Other than that comment, +1 in my inexperienced eyes

File: arch/arch.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// currently running architecture.
+//
+// If the architecture does not map any debian architecture, the
+// GOARCH is returned.
+func UbuntuArchitecture() string {
+	return string(arch)
+}
+
+func goToUbuntuArchitecture() string {
+	switch goarch {
```
Comment (sergiusens) [inline arch/arch.go]: what about `arm64` ? (and maybe the other arches that exist.

File: arch/arch.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+// UbuntuArchitecture returns the debian equivalent architecture for the
+// currently running architecture.
+//
+// If the architecture does not map any debian architecture, the
+// GOARCH is returned.
+func UbuntuArchitecture() string {
+	return string(arch)
+}
+
+func goToUbuntuArchitecture() string {
```
Comment (sergiusens) [inline arch/arch.go]: `goToUbuntuArchichture` was only explained to me when I looked at https://github.com/ubuntu-core/snappy/pull/96/files#diff-fb91c6071604836b89ba2de46c45a56fR36 maybe a clearer name? (maybe it is just me)

File: arch/arch.go (hunk 1 above)
Comment (mvo5) [inline arch/arch.go]: Thanks, I think that makes sense, some functionality in snappy will simply not work (e.g. the generation of the binary and service wrapper scripts) if the architectures can not be mapped so a panic() sounds reasonable here.

File: arch/arch.go (hunk 2 above)
Comment (mvo5) [inline arch/arch.go]: Good point, thanks. I added arm64 and ppc64 now.

File: arch/arch.go (hunk 3 above)
Comment (mvo5) [inline arch/arch.go]: I changed it to ubuntuArchFromGoArch() and added a docstring, does that make it clearer? Happy to change the name again if you have good suggestions.

**Comment by zyga** [comment]
Some tests have failed.

**Comment by chipaca** [comment]
:+1:

Can you take a look at u-d-f and see if it needs tweaking to still build against this?

**Pull request merged**